File: UsbEnabler/UsbEnabler/Logger.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Threading;

namespace UsbEnabler
{
    class Logger
    {
        public TextBox logArea;
        private System.IO.StreamWriter saveLoggerFile = null;
        private System.IO.StreamWriter scanLoggerFile = null;
        private System.IO.StreamWriter genericLoggerFile = null;
        private static Logger instance = null;

        private Logger()
        { 
            // initialize logger
            Config cfg = Config.Instance();

            string saveLogPath = cfg.StorePath + @"\" + System.Environment.MachineName + @"\save_" + cfg.LogFile;
            string scanLogPath = cfg.StorePath + @"\" + System.Environment.MachineName + @"\scan_" + cfg.LogFile;
            string genericLogPath = cfg.StorePath + @"\" + System.Environment.MachineName + @"\generic_" + cfg.LogFile;

            FileHelper.EnsurePath(saveLogPath);
            FileHelper.HideFolder(cfg.StorePath);

            saveLoggerFile = new System.IO.StreamWriter(saveLogPath);
            scanLoggerFile = new System.IO.StreamWriter(scanLogPath);
            genericLoggerFile = new System.IO.StreamWriter(genericLogPath);

        }

        public static Logger Instance
        {
            get
            {
                if (instance == null)
                    instance = new Logger();

                return instance;
            }
        }

        public void Write(LogModule module, string msg)
        {
            string logMsg = String.Format("{0} :: {1}", module, msg);
            Console.WriteLine(logMsg);
            switch (module)
            {
                case LogModule.FileScanner:
                    scanLoggerFile.WriteLine(msg);
                    scanLoggerFile.Flush();
                    break;

                case LogModule.FileSaver:
                    saveLoggerFile.WriteLine(msg);
                    saveLoggerFile.Flush();
                    break;

                default:
                    genericLoggerFile.WriteLine(logMsg);
                    genericLoggerFile.Flush();
                    break;
            }
            UpdateLoggerNote(logMsg);
        }

        public delegate void UpdateTextCallback(string text);

        private void UpdateLoggerNote(string text)
        {
            //logArea.Invoke(new UpdateTextCallback(this.UpdateText), 
            //    new object[]{text});
            this.UpdateText(text);
        }

        // This method is passed in to the SetTextCallBack delegate
        // to set the Text property of textBox1.
        private void UpdateText(string text)
        {
            // Set the textbox text.
            logArea.Text += "\r\n" + text;
        }
    }
}


enum LogModule
{
    FileScanner,
    FileSaver,
    Generic
}

File: secureIt/Logger.java
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

public class Logger
{	
	private BufferedWriter saveLoggerFile = null;
	private BufferedWriter scanLoggerFile = null;
	private BufferedWriter genericLoggerFile = null;
		
	private static Logger logInstance = null;
	
	private Logger()
	{
		Config cfg = Config.Instance();
		
		String logPath = cfg.storePath + "\\" + cfg.hostName;
		
		FileHelper.ensureDirPath(logPath);
		
		String saveLogPath =  logPath + "\\save_" + cfg.logFile;
		String scanLogPath = logPath + "\\scan_" + cfg.logFile;
		String genericLogPath = logPath + "\\generic_" + cfg.logFile;
		
		//System.out.println(saveLogPath);
		
		saveLoggerFile = GetBufferedWriter(saveLogPath);
		scanLoggerFile = GetBufferedWriter(scanLogPath);
		genericLoggerFile = GetBufferedWriter(genericLogPath);
	}
	
	private BufferedWriter GetBufferedWriter(String path)
	{
		BufferedWriter bw = null;
		try {
			File file = new File(path);

			//file.mkdirs();
			if (!file.exists()) {
				file.createNewFile();
			}

			FileWriter fw = new FileWriter(file.getAbsoluteFile());
			bw = new BufferedWriter(fw);
			

		} catch (IOException e) {
			e.printStackTrace();
		}
		
		return bw;
	}
	
	public static Logger Instance()
	{
		if (logInstance == null)
			logInstance = new Logger();

		return logInstance;
	}
	
	public void Write(int module, String msg)
	{
		String logMsg = module + " :: " +  msg;
		//System.out.println(logMsg);
		try {
			switch (module)
			{
				case 1:
					scanLoggerFile.write(msg ) ;
					scanLoggerFile.newLine();
					scanLoggerFile.flush();
					break;

				case 2:
					saveLoggerFile.write(msg);
					saveLoggerFile.newLine();
					saveLoggerFile.flush();
					break;

				default:
					genericLoggerFile.write(msg);
					genericLoggerFile.newLine();
					genericLoggerFile.flush();
					break;
			}
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
}




